app/main.py

```diff
@@ -1,25 +1,98 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: object) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: Any) -> None:
+        if not isinstance(value, int):
+            raise TypeError(f"{self.public_name} should be integer.")
+
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(
+                f"{self.public_name} should not be less than "
+                f"{self.min_amount} and greater than {self.max_amount}."
+            )
+
+        setattr(instance, self.protected_name, value)
 
 
 class Visitor:
-   pass
+    def __init__(
+            self,
+            name: str,
+            age: int,
+            weight: int,
+            height: int
+    ) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(
+            self,
+            age: int,
+            weight: int,
+            height: int
+    ) -> None:
+
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(14, 60)
+    height = IntegerRange(120, 220)
+    weight = IntegerRange(50, 120)
 
 
 class Slide:
-    pass
+    def __init__(
+            self,
+            name: str,
+            limitation_class: [
+                ChildrenSlideLimitationValidator,
+                AdultSlideLimitationValidator
+            ]
```

Comment:
Change your type annotation to indicate that you expect to obtain not instances of `ChildrenSlideLimitationValidator` or `AdultSlideLimitationValidator` but rather a subclass of `SlideLimitationValidator` class
```suggestion
            limitation_class: type[SlideLimitationValidator]
```